cpp/build-support/run_clang_tidy.py

```diff
@@ -0,0 +1,102 @@
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import argparse
+import multiprocessing as mp
+import lintutils
+import sys
+from functools import partial
+import json
+import codecs
+
+sys.stderr = codecs.StreamReader(sys.stderr, 'replace')
+
+def _check_some_files(completed_processes, filenames_json):
+    result = completed_processes[filenames_json]
+    filenames = set(map(lambda s: (s+':').encode('utf-8'), json.loads(filenames_json)))
+    problem_files = set()
+    for line in result.stdout.splitlines():
+        for filename in filenames:
+            if line.startswith(filename):
+                problem_files.add(filename.decode('utf-8'))
+                filenames.remove(filename)
+                break
+    return problem_files, result.stdout
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Runs clang-tidy on all ")
+    parser.add_argument("clang_tidy_binary",
+                        help="Path to the clang-tidy binary")
+    parser.add_argument("compile_commands",
+                        help="compile_commands.json to pass clang-tidy")
+    parser.add_argument("source_dir",
+                        help="Root directory of the source code")
+    parser.add_argument("--fix", default=False,
+                        action="store_true",
+                        help="If specified, will attempt to fix the "
+                        "source code instead of recommending fixes, "
+                        "defaults to %(default)s")
+    parser.add_argument("--quiet", default=False,
+                        action="store_true",
+                        help="If specified, only print errors")
+    arguments = parser.parse_args()
+
+    linted_filenames = [str(path) for path in lintutils.get_sources(arguments.source_dir)]
+
+    if not arguments.quiet:
+        verb = 'Tidying' if arguments.fix else 'Checking'
+        print("\n".join(map(lambda x: "{} {}".format(verb, x),
+                            linted_filenames)))
+
+    cmd = [
+        arguments.clang_tidy_binary,
+        '-p',
+        arguments.compile_commands
+    ]
+    if arguments.fix:
+        cmd.append('-fix')
+        results = lintutils.run_parallel([cmd + some for some in lintutils.chunk(linted_filenames, 16)])
```

Comment:
Can you run flake8 on these file and fix the Python style issues (long lines, etc)?

Reply:
will do